cmd/priv_val_server/main.go

```diff
@@ -32,37 +56,89 @@ func main() {
 		"chainID", *chainID,
 		"privKeyPath", *privValKeyPath,
 		"privStatePath", *privValStatePath,
+		"insecure", *insecure,
+		"withCert", *withCert,
+		"withKey", *withKey,
+		"rootCA", *rootCA,
 	)
 
 	pv := privval.LoadFilePV(*privValKeyPath, *privValStatePath)
 
-	var dialer privval.SocketDialer
-	protocol, address := tmnet.ProtocolAndAddress(*addr)
-	switch protocol {
-	case "unix":
-		dialer = privval.DialUnixFn(address)
-	case "tcp":
-		connTimeout := 3 * time.Second // TODO
-		dialer = privval.DialTCPFn(address, connTimeout, ed25519.GenPrivKey())
-	default:
-		logger.Error("Unknown protocol", "protocol", protocol)
-		os.Exit(1)
+	opts := []grpc.ServerOption{}
+	if !*insecure {
+		certificate, err := tls.LoadX509KeyPair(
+			*withCert,
+			*withKey,
+		)
+		if err != nil {
+			panic(err)
+		}
+
+		certPool := x509.NewCertPool()
+		bs, err := ioutil.ReadFile(*rootCA)
+		if err != nil {
+			panic(fmt.Sprintf("failed to read client ca cert: %s", err))
+		}
+
+		ok := certPool.AppendCertsFromPEM(bs)
+		if !ok {
+			panic("failed to append client certs")
+		}
+
+		tlsConfig := &tls.Config{
+			ClientAuth:   tls.RequireAndVerifyClientCert,
+			Certificates: []tls.Certificate{certificate},
+			ClientCAs:    certPool,
+			MinVersion:   tls.VersionTLS13,
+		}
+
+		creds := grpc.Creds(credentials.NewTLS(tlsConfig))
+		opts = append(opts, creds)
+		logger.Info("SignerServer: Creating security credentials")
+	} else {
+		logger.Error(`SignerServer: You are using an insecure gRPC connection! 
+		Provide a certificate and key to connect securely`)
```

Comment:
```suggestion
		`)
```

if the user supplied `--insecure` option, I'd assume he/she knows what she's doing